backend/infrahub/git/base.py

```diff
@@ -733,6 +774,10 @@ async def fetch(self) -> bool:
 
         return True
 
+    async def get_filtered_remote_branches(self) -> dict[str, BranchInRemote]:
+        remote_branches = self.get_branches_from_remote()
+        return await self.filter_remote_branches(branches=remote_branches)
```

Comment:
As we don't call filter_remote_branches in other locations and always call .get_filtered_remote_branches() I wonder if it's clearer if we remove the filter_remote_branches() method and just have that code inline within get_filtered_remote_branches()?